test/test_opam_cmd.ml

```diff
@@ -0,0 +1,28 @@
+let test_tag_from_archive =
+  let make_test ~archive ~expected =
+    let test_name = "tag_from_archive: " ^ archive in
+    let test_fun () =
+      let actual = Duniverse_lib.Opam_cmd.tag_from_archive archive in 
+      Alcotest.(check (option string)) archive expected actual
+    in
+    (test_name, `Quick, test_fun)
+  in
+  [ make_test ~archive:"git+http://a.com/user/repo" ~expected:(Some "master")
+  ; make_test ~archive:"git+https://a.com/user/repo" ~expected:(Some "master")
+  ; make_test ~archive:"git+https://a.com/user/repo#v0.0.0" ~expected:(Some "v0.0.0")
```

Comment:
what do you think about using `1.2.3` for these version numbers? 0.0.0 seems a bit too symmetric.